samza-core/src/main/java/org/apache/samza/storage/TaskSideInputStorageManager.java

```diff
@@ -68,6 +68,9 @@
   private static final String OFFSET_FILE = "SIDE-INPUT-OFFSETS";
   private static final long STORE_DELETE_RETENTION_MS = TimeUnit.DAYS.toMillis(1); // same as changelog delete retention
   private static final ObjectMapper OBJECT_MAPPER = SamzaObjectMapper.getObjectMapper();
+  private static final TypeReference<HashMap<SystemStreamPartition, String>> OFFSET_TYPE_REFERENCE =
```

Comment:
Nitpick: OFFSETS_TYPE_REFERENCE, since this is for the entire map and not a single offset.